src/jdk.jlink/share/classes/jdk/tools/jlink/internal/runtimelink/JimageDiffGenerator.java

```diff
@@ -128,12 +128,11 @@ private boolean compareStreams(InputStream is1, InputStream is2) {
         byte[] buf2 = new byte[1024];
         int bytesRead1, bytesRead2;
         try (is1; is2) {
-            while ((bytesRead1 = is1.readNBytes(buf1, 0, buf1.length)) != 0) {
+            while (true) {
+                bytesRead1 = is1.readNBytes(buf1, 0, buf1.length);
                 bytesRead2 = is2.readNBytes(buf2, 0, buf2.length);
-                if (bytesRead2 == 0) {
-                    // first input stream didn't return the end of stream
-                    // but the second did?
-                    return false;
+                if (bytesRead1 == 0 || bytesRead2 == 0) {
+                    break; // no more bytes for a stream
```

Comment:
break after test same amount of bytes can remove the test exit the while loop.

Reply:
I went with @archiecobbs suggestion which simplifies this.